### ISSUE TITLE
In the grid encoder, and the spherical encoder, when should calc_grad_inputs be True?

### ISSUE THREAD

**Issue by tym0027** [ashawkey/torch-ngp#100]
Hello,

I was just curious and trying to understand the code, when I noticed that both during training and inference calc_grad_inputs is False because inputs.requires_grad is always false. Should that be the case? If the input never has requires_grad equal to True, then when would you have dy_dx in the encoders be intialized such that dy_dx is "torch.empty(B, L * D * C, device=inputs.device, dtype=embeddings.dtype)" and "torch.empty(B, input_dim * output_dim, dtype=inputs.dtype, device=inputs.device)" in the grid and spherical harmonics encoder respectively?

Thanks

**Comment by ashawkey** [comment]
@tym0027 Hi, if we need to optimize the input coordinates (e.g., in d-nerf), the inputs will have `requires_grad = True` such that the gradient can pass back (e.g, the deformation network in dnerf).
